docs/reference/fields.txt

```diff
@@ -471,6 +471,93 @@ matches strings containing "hello" before a newline, besides strings ending in
 This is because the meaning of ``$`` is different between PCRE and Ruby
 regular expressions.
 
+
+.. _typed-arrays:
+
+Field Type: Typed Arrays
+------------------------
+
+Mongoid supports declaring a typed array field on models. This field will
+enforce the type of the elements in the array by ``demongoizing`` each element
+inserted into the array. To make a typed array field:
+
+.. code:: ruby
+
+  class Band
+    include Mongoid::Document
+    field :members, type: Array(String)
+  end
+
+The members field on the Band class will have class ``Mongoid::StringArray``.
+When an array is assigned to the members field, all of its elements are
+``mongoized`` to Strings. When retrieving the array from the members field, the
+array is ``demongoized`` into a ``Mongoid::StringArray`` and all of its
+elements are ``demongoized`` as well. When subsequently appending to this array,
+the appended elements will also be ``demongoized`` to Strings.
+
+When a typed array field is declared, Mongoid will create a custom class for
+the type: ``Mongoid::<Type>Array``. This class overrides the ``mongoize``,
+``demongoize`` and ``evolve`` methods to convert the elements in the array to
+the array's inner type. This class inherits from the ``Mongoid::TypedArray``
+class.
```

Comment:
Do you intend users to modify these classes? If not I think this needs to be explicitly discouraged here.

Reply:
what do you mean by modify?